### PR TITLE
Make patch operations async for VNGs and VNG Connections

### PR DESCRIPTION
**Pull request by ritwikbasu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [Y] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [Y] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [Y] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [Y] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [Y] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json
Comment (TianoMS) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json]: for 202 response the payload doesn't matter. usually RP just return empty payload.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json
Comment (TianoMS) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json]: same

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json
Comment (ritwikbasu) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json]: For all hybrid networking resources, we return the KVS object being updated from NRP. I have kept the behavior same here as well.

The NRP behavior is designed in the same way as well.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json
Comment (ritwikbasu) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualNetworkGateway.json]: For all hybrid networking resources, we return the KVS object being updated from NRP. I have kept the behavior same here as well.

The NRP behavior is designed in the same way as well.

**Review by TianoMS — APPROVED** [review]
:shipit:

**Comment by MikhailTryakhov** [comment]
@sergey-shandar can you please review & also help to find out why reviewer is not specified automatically?

**Comment by yungezz** [comment]
@MikhailTryakhov the assignment bot is fixed now to auto-assign network PRs. fyi, the problem is caused by network branches not whitelisted.

**Comment by ritwikbasu** [comment]
@sergey-shandar - Can you please review this PR and merge it if you have no comments/suggestions?

**Comment by MikhailTryakhov** [comment]
@yungezz  sorry how to whitelist them? What's a practice to do that next time? Can we add it at the portal during monthly branch creation process?

**Comment by yungezz** [comment]
> @yungezz sorry how to whitelist them? What's a practice to do that next time? Can we add it at the portal during monthly branch creation process?

it's done in bot tool. the fix is not for this month only. the fix is making sure network branches all there, unless you changed branch naming convention.

**Comment by zikalino** [comment]
hmm... for some reason i can't approve this pr/ and merge button also gray...

**Review by zikalino — APPROVED** [review]

**Pull request merged**
